src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -400,4 +400,15 @@ object messages {
            |  ((${nestedRepresentation}))""".stripMargin
     }
   }
+
+  case class MissingReturnType()(implicit ctx:Context) extends Message(14) {
+    val kind = "Syntax"
+    val msg = "missing return type"
+    val explanation =
+      hl"""The return type is missing and needs to be clarified. For example, `area` returns a ${"Double"}:
+        |
+        |trait Shape {
+        |  def area: Double
+        |}""".stripMargin
+  }
```

Comment:
Thanks for the comments. How does this read?
`"""An abstract definition must have a return type. For example:
        |
        |trait Shape {
        |  def area: Double // abstract definition returning a ${"Double"}
        |}"""`